libs/vectorise/include/vectorise/uint/uint.hpp

```diff
@@ -1089,17 +1089,26 @@ inline std::ostream &operator<<(std::ostream &s, UInt<S> const &x)
 
 inline double ToDouble(UInt<256> const &x)
 {
-  static constexpr size_t BITS_IN_UINT64  = sizeof(uint64_t) * 8;
-  static const size_t     MAX_ELEMENT_IDX = x.elements() - 1;
-  size_t                  msw_index       = MAX_ELEMENT_IDX;
-  while (msw_index > 0 && x.ElementAt(msw_index) == uint64_t(0))
+  if (x.TrimmedSize() == 1)
   {
-    --msw_index;
+    return static_cast<double>(x.ElementAt(0));
   }
-  const uint64_t most_significant_word = x.ElementAt(msw_index);
+  const size_t ELEMENTS_PER_WIDE = x.WIDE_ELEMENT_SIZE / x.ELEMENT_SIZE;
+  uint64_t     magnitude         = x.ELEMENTS * x.ELEMENT_SIZE - x.msb();
 
-  return pow(2, static_cast<double>(msw_index * BITS_IN_UINT64)) *
-         static_cast<double>(most_significant_word);
+  const size_t most_sign_byte  = (magnitude - 1) / x.ELEMENT_SIZE;
+  const size_t least_sign_byte = most_sign_byte - (ELEMENTS_PER_WIDE - 1);
```

Comment:
nitpicking, but most_significant_byte/least_significant_byte sound better, sign reminds of the sign bit and is confusing at a first glance.

Reply:
Absolutely agree! I just left a temporary naming here :)